src/se/terhol/pisemka21/ChipLocker.java

```diff
@@ -16,10 +16,40 @@ public ChipKey getLockedBy() {
         return lockedBy;
     }
 
+    public void setLockedBy(ChipKey lockedBy) {
+        this.lockedBy = lockedBy;
+    }
+
     /**
      * @return true if the locker is locked, false otherwise
      */
     public boolean isLocked() {
         return lockedBy != null;
     }
+
+    public void lock(ChipKey key) throws LockerException {
+        if (key == null) {
+            throw new NullPointerException();
+        }
+
+        if (this.isLocked() || key.getLockerLocked() != null && key.getLockerLocked() != this) {
```

Comment:
`key.getLockerLocked() != null` is probably redundant, I think that `this` cannot be `null`

Reply:
I think this is correct, you probably overlooked that it cannot be null either this